### PR TITLE
WIP: Ureq to reqwest

### PR DESCRIPTION
**Pull request by uncomfyhalomacro**
Hello. This is my first time to contribute to this project. As I don't have access with any MacOS machines, I cannot test this change on MacOS so I just added back ureq from this commit 88a98a0eb9f364c67aa1ea3058820d04523a2cbd . For now, I am not sure if this will work on Linux (for the most part I tested it and it worked well on my openSUSE machine).

### CODE REVIEW

**Comment by uncomfyhalomacro** [comment]
Hmm openssl build errors?

**Comment by uncomfyhalomacro** [comment]
Commit https://github.com/JuliaLang/juliaup/pull/613/commits/11405a1e2482fa435a69faf4c8ae0d3513db80d2 changes to only check non-windows to test if it builds in macos correctly. however tests are cancelled because of a build failure.

**Comment by uncomfyhalomacro** [comment]
seems i just need to disable features of reqwest.

File: Cargo.toml [hunk 1]
```diff
@@ -45,12 +45,11 @@ indoc = "1.0.7"
...
-[target.'cfg(target_os = "macos")'.dependencies]
-ureq = { version = "2", features = ["native-tls"] }
-native-tls = "0.2.10"
+[target.'cfg(not(any(target_os = "windows", target_env = "musl")))'.dependencies]
+reqwest = { version = "0.11.16",  default-features = false, features = ["blocking", "native-tls"] }
 
-[target.'cfg(all(not(target_os = "macos"),not(windows)))'.dependencies]
-ureq = { version = "2", features = ["native-certs"] }
+[target.'cfg(all(target_os = "linux", target_env = "musl"))'.dependencies]
+reqwest = { version = "0.11.16",  default-features = false, features = ["blocking", "native-tls-vendored"] }
```
Comment (davidanthoff) [inline Cargo.toml:52]: I think you want the entire block for `reqwest` look like this:
```
[target.'cfg(target_os = "macos")'.dependencies]
reqwest = { version = "0.11", default-features = false, features = ["blocking", "native-tls"] }

[target.'cfg(all(not(target_os = "macos"),not(windows)))'.dependencies]
reqwest = { version = "0.11", default-features = false, features = ["blocking", "rustls-tls-native-roots"] }
```

I think that would be equivalent to how we handle this right now in terms of using certificates etc. I think you then also don't need to worry about installing `libssl` etc.

File: .github/workflows/test.yml
Comment (davidanthoff) [inline .github/workflows/test.yml:1]: I think you can remove all changes in this file if you adopt my suggestion below from `Cargo.toml`.

File: src/operations.rs [hunk 2]
```diff
@@ -48,7 +48,7 @@ where
 }
 
 #[cfg(not(windows))]
-fn get_proxy(url: &str) -> Option<Result<ureq::Proxy>> {
+fn get_proxy(url: &str) -> Option<Result<reqwest::Proxy>> {
```
Comment (davidanthoff) [inline src/operations.rs:51]: Do we even need this, or does `reqwest` handle proxies fully automatically?

**Review by davidanthoff — COMMENTED** [review]
Left a couple of comments. Let me know if you want to address them, otherwise I might also just push to this branch?

File: Cargo.toml (hunk 1 above)
Comment (davidanthoff) [reply Cargo.toml]: I just pushed a commit that fixes this.

File: .github/workflows/test.yml
Comment (davidanthoff) [reply .github/workflows/test.yml:1]: I also did this.

File: src/operations.rs (hunk 2 above)
Comment (davidanthoff) [reply src/operations.rs]: I removed all the manual proxy handling, my understanding is that reqwest does this transparently in the background.

**Comment by uncomfyhalomacro** [comment]
> Left a couple of comments. Let me know if you want to address them, otherwise I might also just push to this branch?

Hello. thanks for reviewing and merging my PR. I didn't notice as I got busy as well. I haven't simplified it as I was just trying to not change the logic at first.

**Comment by davidanthoff** [comment]
> EDIT: Seems most of the code changes are still yours. oooof

Well, I just deleted stuff ;) Your initial work was super helpful, much appreciated!

**Pull request merged**